src/Reducers/gameReducer.tsx

```diff
@@ -0,0 +1,83 @@
+import { ActionType, StateType } from "../types";
+import { actionNames, COLORS } from "../constants";
+import {
+  generateAnswer,
+  generateN,
+  getOutput,
+  getResult,
+  generateStyles,
+  generateOutput,
+} from "../Utils/functions";
+import { constraints } from "../constants";
+
+const gameReducer = (state: StateType, action: ActionType) => {
+  const { colorIndex, index } = action.payload;
+  const { activeIndex, styles, input, selectedColor, output } = state;
+
+  switch (action.type) {
+    case actionNames.NEW_GAME: {
+      return {
+        ...state,
+        selectedColor: 0,
+        activeIndex: 0,
+        styles: generateStyles(
+          constraints.NO_OF_CHANCES,
+          constraints.NO_OF_GUESSES
+        ),
+        input: generateN(constraints.NO_OF_GUESSES, 7),
+        answer: generateAnswer(input.length),
+        output: generateOutput(
+          constraints.NO_OF_CHANCES,
+          constraints.NO_OF_GUESSES,
+          0
+        ),
+        isWinner: false,
+        isOver: false,
+      };
+    }
+
+    case actionNames.SELECT_COLOR: {
+      return {
+        ...state,
+        selectedColor: colorIndex,
+      };
+    }
+
+    case actionNames.SET_COLOR: {
+      let newStyles = styles;
```

Comment:
ref of newStyle and style is same here
so indirectly you are updating style as well

look for cloning/copy of object/array

Reply:
deep cloned the style